libs/common/src/test/java/org/opensearch/common/round/RoundableTests.java

```diff
@@ -8,18 +8,60 @@
 
 package org.opensearch.common.round;
 
+import org.opensearch.common.SuppressForbidden;
 import org.opensearch.test.OpenSearchTestCase;
 
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.InvocationTargetException;
+
 public class RoundableTests extends OpenSearchTestCase {
 
-    public void testFloor() {
-        int size = randomIntBetween(1, 256);
-        long[] values = new long[size];
-        for (int i = 1; i < values.length; i++) {
-            values[i] = values[i - 1] + (randomNonNegativeLong() % 200) + 1;
+    public void testBidirectionalLinearSearcher() {
+        assertRounding(BidirectionalLinearSearcher::new);
+    }
+
+    public void testBinarySearcher() {
+        assertRounding(BinarySearcher::new);
+    }
+
+    @SuppressForbidden(reason = "Reflective construction of BtreeSearcher since it's not supported below Java 20")
+    public void testBtreeSearcher() {
+        RoundableSupplier supplier;
+
+        try {
+            Class<?> clz = MethodHandles.lookup().findClass("org.opensearch.common.round.BtreeSearcher");
```

Comment:
We are back to where we started :( That is not the way we should be writing tests